### PR TITLE
Validation of isArchived

### PR DESCRIPTION
**Pull request by D-Raoul**

### CODE REVIEW

File: app/Espo/Core/Templates/Repositories/Hierarchy.php [hunk 1]
```diff
@@ -398,6 +398,35 @@ public function getParentRecord(string $id): array
...
+        return !empty($this->getMetadata()->get(['scopes', $entity->getEntityType(), 'hasArchive']));
+    }
+
+
+    protected function validateIsArchived(Entity $entity): void
+    {
+        $fieldName = 'isArchived';
+        if ($entity->isAttributeChanged($fieldName) && $entity->get($fieldName)==true) {
+            // search all childs
+            if (!empty($entity->get('childrenIds'))) {
```
Comment (rratsun) [inline app/Espo/Core/Templates/Repositories/Hierarchy.php:412]: There are no needs to find deeper than one level. Also, change $entity->get('childrenIds') to $children = $entity->get('children'), then check every item of collection and exit at first not archive value.

Comment (D-Raoul) [reply app/Espo/Core/Templates/Repositories/Hierarchy.php]: Fixed

File: app/Espo/Core/Templates/Repositories/Hierarchy.php [hunk 2]
```diff
@@ -398,6 +398,32 @@ public function getParentRecord(string $id): array
...
+
+
+    protected function validateIsArchived(Entity $entity): void
+    {
+        $fieldName = 'isArchived';
+        if ($entity->isAttributeChanged($fieldName) && $entity->get($fieldName) == true) {
+            // search all childs
+            $hasNonArchivedChildren = false;
+            foreach ($entity->get('children') as $child) {
+                if ($child->get('is_archived') == false) {
```
Comment (rratsun) [inline app/Espo/Core/Templates/Repositories/Hierarchy.php:414]: Why ```is_archived``` in underscore? Should be isArchived

Comment (D-Raoul) [reply app/Espo/Core/Templates/Repositories/Hierarchy.php]: Fixed

**Pull request merged**